src/emqx_cm.erl

```diff
@@ -447,17 +457,27 @@ handle_call(Req, _From, State) ->
 handle_cast({registered, {ClientId, ChanPid}}, State = #{chan_pmon := PMon}) ->
     PMon1 = emqx_pmon:monitor(ChanPid, ClientId, PMon),
     {noreply, State#{chan_pmon := PMon1}};
-
 handle_cast(Msg, State) ->
     ?LOG(error, "Unexpected cast: ~p", [Msg]),
     {noreply, State}.
 
 handle_info({'DOWN', _MRef, process, Pid, _Reason}, State = #{chan_pmon := PMon}) ->
+    ?tp(emqx_cm_process_down, #{pid => Pid, reason => _Reason}),
     ChanPids = [Pid | emqx_misc:drain_down(?BATCH_SIZE)],
     {Items, PMon1} = emqx_pmon:erase_all(ChanPids, PMon),
+    lists:foreach(
+      fun({ChanPid, ClientID}) ->
+              Chan = {ClientID, ChanPid},
+              case ets:lookup(?CHAN_INFO_TAB, Chan) of
+                  [{Chan, #{conn_state := connected}, _}] ->
+                      decrement_connected_client_count(ChanPid);
+                  _ ->
+                      ok
+              end
```

Comment:
I guess there is no need to check anything, just call `mark_disconnected(ChanPid)`

Reply:
You are right, now that an ETS set is used, there is no longer a need to check the state.